lldb/include/lldb/API/SBTarget.h

```diff
@@ -986,6 +987,42 @@ class LLDB_API SBTarget {
 
   lldb::SBMutex GetAPIMutex() const;
 
+  /// Register a scripted frame provider for this target.
+  /// If a scripted frame provider with the same name is already registered on
+  /// this target, it will be overwritten.
+  ///
+  /// \param[in] class_name
+  ///     The name of the Python class that implements the frame provider.
+  ///
+  /// \param[in] args_dict
+  ///     A dictionary of arguments to pass to the frame provider class.
+  ///
+  /// \param[in] thread_filter
+  ///     The list of threads where to register the frame provider class. If
+  ///     empty, apply scripted frame provider to all threads.
+  ///
+  /// \return
+  ///     An error object indicating success or failure.
+  lldb::SBError
+  RegisterScriptedFrameProvider(const char *class_name,
+                                lldb::SBStructuredData args_dict,
+                                lldb::SBThreadCollection thread_filter = {});
+
+  /// Remove a scripted frame provider from this target by name.
+  ///
+  /// \param[in] provider_name
+  ///     The name of the frame provider class to remove.
+  ///
+  /// \return
+  ///     An error object indicating success or failure.
+  lldb::SBError RemoveScriptedFrameProvider(const char *provider_name);
+
+  /// Clear all scripted frame providers for this target.
+  ///
+  /// \return
+  ///     An error object indicating success or failure.
+  lldb::SBError ClearScriptedFrameProvider();
```

Comment:
Should this then also be `ClearSyntheticFrameProvider(s)`?

Reply:
As mentioned here  https://github.com/llvm/llvm-project/pull/161870#discussion_r2505912411, the target only hold `ScriptedFrameProviderDescriptor` internally so I think calling Scripted is what makes the most sense.